ray-operator/controllers/ray/common/pod.go

```diff
@@ -711,6 +711,14 @@ func addEmptyDir(container *v1.Container, pod *v1.Pod, volumeName string, volume
 	if checkIfVolumeMounted(container, pod, volumeMountPath) {
 		return
 	}
+
+	// 0) Skip adding the volume if it is already in the list of volumes
+	for _, volume := range container.VolumeMounts {
+		if volume.Name == volumeName && volume.MountPath == volumeMountPath {
```

Comment:
I have some concerns about this line. MountPath needs to be unique. Hence, if users specify a volume with a unique name but the same MountPath (`/tmp/ray` or `/dev/shm`), Kubernetes apiserver will report error. 

Take the following YAML file as an example, when you use `kubectl apply` to apply the YAML file, Kubernetes apiserver will report `The Pod "test-pd" is invalid: spec.containers[0].volumeMounts[1].mountPath: Invalid value: "/cache": must be unique`.

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: test-pd
spec:
  containers:
  - image: ubuntu:latest
    command:
    - "sleep"
    - "604800"
    name: test-container
    volumeMounts:
    - mountPath: /cache
      name: cache-volume
    - mountPath: /cache
      name: cache-volume-2
  volumes:
  - name: cache-volume
    emptyDir: {}
  - name: cache-volume-2
    emptyDir: {}
```

Reply:
In my opinion, the logic should be

```go
	// 0) Skip adding the volume if it is already in the list of volumes
	for _, volume := range container.VolumeMounts {
		if volume.MountPath == volumeMountPath {
                     return
                }
        }
```

Reply:
This is a good point. To be make the check even stricter, we can avoid duplicating either name or path. Also, the variable iteration variable should be `volumeMount`.

I suggest
```go
	// 0) Skip adding the volume if it is already in the list of volumes
	for _, volumeMount := range container.VolumeMounts {
		if volumeMount.Name == volumeName || volumeMount.MountPath == volumeMountPath {
			return
		}
	}
```

Reply:
(1) Duplicate `volumeMount.Name` is allowed in Kubernetes. That is, two MountPaths are the same directory. This is weird, but files written to both MountPaths can be persisted. 

(2) If we skip the duplicate `volumeMount.Name`, it is possible to have no `volumeMount` for `/tmp/ray` or `/dev/shm`. Hence, some important data may not be persisted.


Both (1) and (2) may cause users confusing. For me, I will report an error if any duplicate `volumeMount.Name` or `volumeMount.MountPath` exists, but this may be too aggressive.

Reply:
Makes sense, de-duping mount path seems enough.

Reply:
updated